frontend/leaderboard/src/Sheets.re

```diff
@@ -57,6 +58,50 @@ let getRange = (client, sheetsQuery, cb) => {
   });
 };
 
+let updateRange = (client, sheetsUpdate, cb) => {
+  let sheets = GoogleSheets.sheets({version: "v4", auth: client});
+
+  GoogleSheets.update(sheets, sheetsUpdate, (~error, ~res) => {
+    switch (Js.Nullable.toOption(error)) {
+    | None => cb(Ok(res.data.values))
+    | Some(error) => cb(Error(error))
+    }
+  });
+};
+
+let createPublickeyUsernameMap = (pointsMap, sheetsData) => {
+  sheetsData
+  |> Array.fold_left(
+       (map, user) => {
+         // Check if the publickey exists in the pointsMap
+         StringMap.mem(user[0], pointsMap)
```

Comment:
We can remove this check and then we don't need the pointsMap to be passed to this function, just always add the username mapping

Reply:
Fixed.